docs/workspaces/index.md

```diff
@@ -154,6 +152,45 @@ A minimal `question.html` for an externally graded workspace should look somethi
 </pl-submission-panel>
 ```
 
+### Creating files in the workspace home directory
+
+Workspace questions can optionally include a `workspace/` subdirectory within the regular [PrairieLearn question directory structure](../question.md#directory-structure). If this `workspace/` subdirectory exists, its contents will be copied into the home directory of the student's workspace container, as configured in the `home` setting in `info.json`.
+
+Questions using workspaces can also be randomized, i.e., include files that contain random and dynamic content. This is done using [the `server.py` file in the question directory](../question.md#question-serverpy). In addition to other random parameters that can be created for the question page itself, the `_files` parameter can also be set, containing an array of files to be created. Each element of the array must include a `name` property, containing the file name (which can include a path with directories), and either a `contents` property, containing the contents of the file, or a `questionFile` property, pointing to an existing file in a different location in the question directory. For example:
```

Comment:
I like the flexibility of this mechanism, but I'm a bit worried that it has a lot of overhead for most users. Could we also support a `workspaceTemplates/` directory that contains files which we will process with mustache (just like `question.html`, with the same `data` available) and copy into the workspace? I'm thinking the precedence would be (1) files from `workspace/`, (2) files files from `workspaceTemplates/`, and finally (3) files from `_files`.

Reply:
I'm fine with that. Is it ok if I move it to a future PR?

Reply:
Well, yeah, I guess, but I'm worried it will never happen 😄. Could I lean on you to rush it into this PR? I know I've already delayed this PR a lot so I won't block merging over this request, but it would be so nice to see this feature added (and a News item!).

Reply:
I'll start on it soon, as soon as this is merged. See #7089. I assume the news item should go to that PR, and include both dynamic files and wildcards in #6982?